docs/design.md

```diff
@@ -0,0 +1,144 @@
+# GCP Ingestion Design
+
+This document specifies the design for GCP Ingestion as a whole.
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+
+- [Architecture Diagram](#architecture-diagram)
+- [Data Flow](#data-flow)
+  - [Notes](#notes)
+- [Design Decisions](#design-decisions)
+  - [Kubernetes and PubSub](#kubernetes-and-pubsub)
+  - [Different topics for "raw" and "validated"](#different-topics-for-raw-and-validated)
+  - [BigQuery](#bigquery)
+  - [Saving as JSON](#saving-as-json)
+  - [Use destination tables?](#use-destination-tables)
+  - [Use views for user-facing data?](#use-views-for-user-facing-data)
+- [Known Issues](#known-issues)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Architecture Diagram
+
+![alt text](architecture.svg "Architecture Diagram")
```

Comment:
I think the color coding in the diagram is really useful, so you could put some text after the diagram to explicitly call out what the colors mean and link to the subprojects where the relevant code lives.

Reply:
I added a diagram describing the color meanings right before this